### ISSUE TITLE
Unresolved external symbol when compiling for 32bit windows

### ISSUE THREAD

**Issue by MidasLamb** [wasmerio/wasmer#2414]
### Describe the bug
I'm compiling for windows 32 bit from a 64bit machine by doing 
```sh
cargo +stable-i686-pc-windows-msvc --target i686-pc-windows-msvc r
```

And I get following error: 
```sh
= note: libwasmer_vm-dc29bfe30745a26c.rlib(wasmer_vm-dc29bfe30745a26c.wasmer_vm.7cxdr9x9-cgu.15.rcgu.o) : error LNK2001: unresolved external symbol ___rust_probestack
          libwasmer_vm-dc29bfe30745a26c.rlib(wasmer_vm-dc29bfe30745a26c.wasmer_vm.7cxdr9x9-cgu.8.rcgu.o) : error LNK2001: unresolved external symbol ___rust_probestack
          C:\Users\Midas\Desktop\projects\dbs2\server\target\i686-pc-windows-msvc\debug\deps\server.exe : fatal error LNK1120: 1 unresolved externals
```

### Steps to reproduce
Have the following program: 
```rust
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    use wasmer::{imports, Instance, Module, Store, Value};
    // some other unrelated stuff, only having the use above is sufficient.
   println!("Hello world")
}
```
while having this in the dependencies in Cargo.toml
```toml
wasmer = {git = "https://github.com/wasmerio/wasmer.git", branch = "master", default-features = false, features = ["default-cranelift", "default-universal", "wat"]}
```
and try to run it with:
```sh
cargo +stable-i686-pc-windows-msvc --target i686-pc-windows-msvc r
```

### Expected behavior
Compilation succeeds and be able to run it.

### Actual behavior
Compilation fails during linking (`cargo check` succeeds)

I tried to solve this myself but I have no idea why this happens so I'm stuck :/

**Comment by Hywan** [comment]
That's because we don't support Windows 32bits in our probestack module. It fallbacks to the last `else` block: https://github.com/wasmerio/wasmer/blob/144300e0bc51d01290680a46cd30a6ca76ed0359/lib/vm/src/probestack.rs#L20-L53 hence the error.

We could add [Windows Server 2016](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2016-Readme.md) to our CI ([the identifier is `windows-2016` in Github Actions](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)), for what I understand it's running on 32bits.

After that, we could try to patch `wasmer-vm` to support Windows on 32bits. I believe Windows 32bits (`x86`) is also using the `__chkstk` “routine”, just like the 64bits version (`x64` as they name it). On the short term, we can also just try to patch `wasmer-vm` and if it works for you, then good, we could merge it, and then add Windows 32bits in our CI.

**Comment by Hywan** [comment]
Note that we never tried to support Windows 32bits, so it's very likely that more patches could be needed.

**Comment by MidasLamb** [comment]
@Hywan, thanks very much for the insight, I just tried to patch it and it seems to be `_chkstk` (with only one leading underscore), I'll try to get it running now and if it works, I'll submit a PR for it!

**Comment by MidasLamb** [comment]
It gets me one step further, but now I get the following error
```sh
thread 'main' panicked at 'construct Cranelift ISA for triple: Unsupported', C:\Users\Midas\Desktop\projects\wasmer\lib\compiler-cranelift\src\config.rs:72:45
```
which seems to be caused by cranelift itself, so my next best bet is to switch `default-cranelift` to one of the other two options (al though those also give me errors at the moment, but the llvm one seems to be a config error on my end)
I'll tidy up my code change and submit a PR for it already

**Comment by syrusakbary** [comment]
I would recommend using llvm. Singlepass is not designed for 32-bit systems

**Comment by MidasLamb** [comment]
@syrusakbary, will try, but I don't have much experience with LLVM and I'm yet to be able to make the compiler happy with having a version of LLVM on my system that it can find.

**Comment by MidasLamb** [comment]
Resolved in #2415

**Comment by w4454962** [comment]
@MidasLamb   Can you help me compile the latest SDK for 32-bit Windows? I'm not very familiar with rust and LLVM, but I really need the x86-32 SDK
